mindmaps-graql/src/main/java/io/mindmaps/graql/internal/analytics/StdMapReduce.java

```diff
@@ -0,0 +1,113 @@
+package io.mindmaps.graql.internal.analytics;
+
+import io.mindmaps.concept.ResourceType;
+import io.mindmaps.util.Schema;
+import org.apache.commons.configuration.Configuration;
+import org.apache.tinkerpop.gremlin.process.computer.KeyValue;
+import org.apache.tinkerpop.gremlin.structure.Graph;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+import org.apache.tinkerpop.gremlin.util.iterator.IteratorUtils;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+import static io.mindmaps.graql.internal.analytics.Analytics.getVertexType;
+
+public class StdMapReduce extends MindmapsMapReduce<Map<String, Number>> {
+    public static final String MEMORY_KEY = "std";
+    public static final String SELECTED_DATA_TYPE = "SELECTED_DATA_TYPE";
+
+    public static final String COUNT = "C";
+    public static final String SUM = "S";
+    public static final String SQUARE_SUM = "SM";
+
+    private String resourceDataType = null;
+
+    public StdMapReduce() {
+    }
+
+    public StdMapReduce(Set<String> selectedTypes, Map<String, String> resourceTypes) {
+        this.selectedTypes = selectedTypes;
+        resourceDataType = resourceTypes.get(selectedTypes.iterator().next());
+    }
+
+    @Override
+    public void storeState(final Configuration configuration) {
```

Comment:
Use the persistentProperties map instead of overwriting storeState, that's its purpose - anything runtime property that isn't generic.